assets/app/views/create/fromimage.html

```diff
@@ -19,7 +19,14 @@
                 <div style="margin-bottom: 15px;">
                   <div class="form-group form-group-lg">
                    <label for="appname" class="required">Name</label>
-                   <div ng-class="{'has-error': ((form.appname.$invalid && form.appname.$touched) || nameTaken)}">
+                   <!--
+                     Only validate the name pattern and minlength if the field has both been changed and blurred.  This
+                     avoids a lot of problems clicking on the try it / more options links below.  Don't use $touched,
+                     which causes validation error messages to flash when typing in the field after clicking one of the
+                     links. Show the required error if the user enters and deletes the value, even if the field still
+                     has focus.
+                   -->
+                   <div ng-class="{'has-error': (form.appname.$error.required && form.appname.$dirty) || (form.appname.$invalid && validateName) || nameTaken}">
```

Comment:
the meaning of validateName confused me at first because it reads as an action. maybe shouldValidateName ?